`p2p_invalid_messages.py` failed on Travis and also on my machine with macOS 10.14.3:

```
2019-01-31T16:36:14.813000Z TestFramework (INFO): Sending a bunch of large, junk messages to test memory exhaustion. May take a bit...
2019-01-31T16:36:18.519000Z TestFramework (INFO): Waiting for node to drop junk messages.
Fatal write error on socket transport
protocol: <test_framework.mininode.P2PDataStore object at 0x10ebcb8d0>
transport: <_SelectorSocketTransport fd=11 read=polling write=<idle, bufsize=0>>
Traceback (most recent call last):
  File "/Users/bitcoin/.pyenv/versions/3.4.9/lib/python3.4/asyncio/selector_events.py", line 701, in write
    n = self._sock.send(data)
OSError: [Errno 41] Protocol wrong type for socket
```

The problem goes away when I rebase on master, which includes #15246. I assume merging will have the same effect.

--- a/test/functional/test_framework/test_node.py
+++ b/test/functional/test_framework/test_node.py
@@ -402,6 +402,14 @@ class TestNodeCLIAttr:
     def get_request(self, *args, **kwargs):
         return lambda: self(*args, **kwargs)
 
+def arg_to_cli(arg):
+    if isinstance(arg, bool):
+        return str(arg).lower()
+    elif isinstance(arg, dict) or isinstance(arg, list):
+        return json.dumps(arg)
+    else:
+        return str(arg)
+
 class TestNodeCLI():
     """Interface to bitcoin-cli for an individual node"""
 
@@ -433,8 +441,8 @@ class TestNodeCLI():
 
     def send_cli(self, command=None, *args, **kwargs):
         """Run bitcoin-cli command. Deserializes returned string as python object."""
-        pos_args = [str(arg).lower() if type(arg) is bool else str(arg) for arg in args]
-        named_args = [str(key) + "=" + str(value) for (key, value) in kwargs.items()]
+        pos_args = [arg_to_cli(arg) for arg in args]
+        named_args = [str(key) + "=" + arg_to_cli(value) for (key, value) in kwargs.items()]
         assert not (pos_args and named_args), "Cannot use positional arguments and named arguments in the same bitcoin-cli call"
         p_args = [self.binary, "-datadir=" + self.datadir] + self.options
         if named_args:

